spring-integration-core/src/test/java/org/springframework/integration/support/management/observation/IntegrationObservabilityZipkinTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management.observation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.annotation.BridgeTo;
+import org.springframework.integration.annotation.EndpointId;
+import org.springframework.integration.annotation.Poller;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.channel.interceptor.ObservationPropagationChannelInterceptor;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.config.EnableIntegrationManagement;
+import org.springframework.integration.config.GlobalChannelInterceptor;
+import org.springframework.integration.support.MutableMessage;
+import org.springframework.integration.support.MutableMessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.support.ChannelInterceptor;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.core.tck.MeterRegistryAssert;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+import io.micrometer.tracing.Span;
+import io.micrometer.tracing.test.SampleTestRunner;
+import io.micrometer.tracing.test.simple.SpansAssert;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 6.0
+ */
+public class IntegrationObservabilityZipkinTests extends SampleTestRunner {
+
+	@Override
+	public TracingSetup[] getTracingSetup() {
+		return new TracingSetup[]{ TracingSetup.IN_MEMORY_BRAVE, TracingSetup.ZIPKIN_BRAVE };
```

Comment:
Why are you leaving out OTel ? BTW the Wavefront setup will be automatically ignored cause you didn't provide any tokens.

Do you have a screenshot from Zipkin? I'm super interested in how it looks like :)

Reply:
Why should I care about OTel?
It is really not Spring Integration responsibility to check the target tracing protocol.
I have even excluded OTel and Wavefront dependencies: it is not my project responsibility to test Observation low-level libraries.
I think I just have to believe to your asserts and that's it 😄

Reply:
> Why should I care about OTel?

Because we're providing 2 tracers for this `SampleTestRunner` and we want to be sure that using both of them things are looking good.

>  It is really not Spring Integration responsibility to check the target tracing protocol.

You can also just write unit tests with mocks - it's all about being more certain that things are working fine

>  I have even excluded OTel and Wavefront dependencies: it is not my project responsibility to test Observation low-level libraries.

`SampleTestRunner` is about giving you more confidence that things are looking good with tracing visualization tools. If I was a Spring Integration project maintainer then I would like to be pretty sure that things are looking good regardless of the used tracer. Also I would like to be sure that things are looking good in Wavefront (have you checked that?)

Reply:
Well, this sounds like testing a `RestTemplate` against different HTTP client impls, which is really not my project responsibility where I should worry proper `RestTemplate` calls from my code. How `RestTemplate` interacts with a provided HTTP client is really outside of my project scope.
OTel, Wavefront, Zipkin - all of these are new for me and I need to read about them more.
Meanwhile I still don't see what and how I would fix in  my code if something is not OK with those techs. Just because I don't do anything from my code to affect it somehow.
Doesn't look like it is similar to RDBMS where each vendor may have its own specific SQL extensions we should care about.

If you still insist on enabling them, I'll revert my restrictions, but, please, be my guest to help me to determine "that things are looking good" 😄